prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -140,6 +138,11 @@ class ReportFunction(
         val httpStatus: HttpStatus =
             try {
                 val options = Options.valueOfOrNone(optionsText)
+                if (Options.isDeprecated(options)) {
```

Comment:
Consider changing this to `options.isDeprecated` instead of using a companion object since the operation will always be done on an instance of that enum